### PR TITLE
Add integration test for rabbitmq receiver

### LINKED ISSUE #10865: [receiver/rabbitmq] Add integration test
The task is essentially to create an integration test that:
1. Starts an rabbitmq docker container
2. Initializes the rabbitmqreceiver component from it's factory
3. Scrapes metrics from the docker container
4. Validates that the expected metrics are scraped (using the [scrapertest package](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/scrapertest))

As far as examples go, there are a few components out there that have integration tests using the same pattern.
For instance:

[Mysql receiver integration test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8eec12f2688309fe4bf0b97ad73128ab12d72861/receiver/mysqlreceiver/integration_test.go)
[Nginx receiver integration test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8eec12f2688309fe4bf0b97ad73128ab12d72861/receiver/nginxreceiver/integration_test.go)

_Adapted from post by @BinaryFissionGames in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10165#issuecomment-1133257413_

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** Add integration test for rabbitmq receiver

resolved #10865

### CODE REVIEW

**Comment by Frapschen** [comment]
@djaglowski revew pls :)

File: receiver/rabbitmqreceiver/testdata/integration/scripts/setup.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -e
+
+USER='otel'
+ROOT_PASS='otel'
```
Comment (djaglowski) [inline receiver/rabbitmqreceiver/testdata/integration/scripts/setup.sh:6]: nit: Can we set these to different values, to ensure we are not crossing them internally?
```suggestion
USER='otelu'
ROOT_PASS='otelp'
```

File: receiver/rabbitmqreceiver/testdata/integration/scripts/setup.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+rabbitmqctl add_user ${USER} ${ROOT_PASS} > /dev/null
+
+echo "Configuring ${USER} permissions. . ."
+rabbitmqctl set_user_tags otel administrator > /dev/null
+rabbitmqctl set_permissions -p /  ${USER} ".*" ".*" ".*" > /dev/null
+
+echo "create exchange and queue. . ."
+rabbitmqadmin -u ${USER} -p ${ROOT_PASS} declare exchange --vhost=/ name=some_exchange type=direct > /dev/null
+rabbitmqadmin -u ${USER} -p ${ROOT_PASS} declare queue --vhost=/ name=some_outgoing_queue durable=true > /dev/null
+rabbitmqadmin -u ${USER} -p ${ROOT_PASS} declare binding source="some_exchange" destination_type="queue" destination="some_outgoing_queue" > /dev/null
```
Comment (djaglowski) [inline receiver/rabbitmqreceiver/testdata/integration/scripts/setup.sh:18]:
```suggestion
rabbitmqadmin -u ${USER} -p ${ROOT_PASS} declare binding source="some_exchange" destination_type="queue" destination="some_outgoing_queue" > /dev/null

```

File: receiver/rabbitmqreceiver/testdata/integration/Dockerfile.rabbitmq.3_9 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FROM rabbitmq:3.9-management
+
+COPY scripts/setup.sh /setup.sh
+RUN chmod +x /setup.sh
+
+EXPOSE 15672
```
Comment (djaglowski) [inline receiver/rabbitmqreceiver/testdata/integration/Dockerfile.rabbitmq.3_9:6]:
```suggestion
EXPOSE 15672

```

**Review by djaglowski — COMMENTED** [review]
Overall this looks good. A few minor comments though.

File: receiver/mysqlreceiver/README.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ This receiver queries MySQL's global status and InnoDB tables.
 
 This receiver supports MySQL version 8.0
 
-Collecting most metrics requires the ability to execute `SHOW GLOBAL STATUS`. The `buffer_pool_size` metric requires access to the `information_schema.innodb_metrics` table. Please refer to [setup.sh](./testdata/integration/scripts/setup.sh) for an example of how to configure these permissions. 
+Collecting most metrics requires the ability to execute `SHOW GLOBAL STATUS`. The `buffer_pool_size` metric requires access to the `information_schema.innodb_metrics` table. Please refer to [setup.sh](testdata/integration/scripts/setup.sh) for an example of how to configure these permissions.
```
[the remark this answers is not in the document]
Comment (Frapschen) [reply receiver/mysqlreceiver/README.md]: done

File: receiver/rabbitmqreceiver/integration_test.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+			require.NoError(t, container.Terminate(context.Background()))
+		}()
+		hostname, err := container.Host(context.Background())
+		require.NoError(t, err)
+
+		f := NewFactory()
+		cfg := f.CreateDefaultConfig().(*Config)
+		cfg.Endpoint = fmt.Sprintf("http://%s:15672", hostname)
+		cfg.Username = "otel" //otel guest
+		cfg.Password = "otel"
```
Comment (djaglowski) [inline receiver/rabbitmqreceiver/integration_test.go:64]: Don't these need to be updated as well? I wonder if we'll see more metrics if these are updated.

File: receiver/rabbitmqreceiver/testdata/integration/expected.3_9.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                  "key": "rabbitmq.vhost.name",
+                  "value": {
+                     "stringValue": "/"
+                  }
+               }
+            ]
+         },
+         "scopeMetrics": [
+            {
+               "metrics": [
```
Comment (djaglowski) [inline receiver/rabbitmqreceiver/testdata/integration/expected.3_9.json:28]: Only one metric? Any idea why?

File: receiver/rabbitmqreceiver/integration_test.go (hunk 5 above)
Comment (Frapschen) [reply receiver/rabbitmqreceiver/integration_test.go]: ok

File: receiver/rabbitmqreceiver/testdata/integration/expected.3_9.json (hunk 6 above)
Comment (Frapschen) [reply receiver/rabbitmqreceiver/testdata/integration/expected.3_9.json:28]: @djaglowski I have tried mock some message push and get, I still get one metric, I think the receiver only scrape the current status of the queue, as one of  metric's  description  ——` The number of consumers currently reading from the queue. `   

as for the metric `rabbitmq.message.published` will get right value. the same as `rabbitmq.message.curren`

**Comment by djaglowski** [comment]
@Frapschen, thank you. This looks good to me. Please resolve conflicts

**Comment by Frapschen** [comment]
@djaglowski resolved. I want to apply become a member of opentelemetry community,can you sponsored me?

**Comment by djaglowski** [comment]
👍 Happy to sponsor you @Frapschen. Thanks for all your contributions so far.

**Comment by djaglowski** [comment]
Looks like some new CI failures with that most recent push

**Comment by Frapschen** [comment]
@djaglowski fixed

**Comment by djaglowski** [comment]
```
=== CONT  TestRabbitmqIntegration/Running_rabbitmq_3.9
    integration_test.go:93: 
        	Error Trace:	integration_test.go:93
        	            				integration_test.go:50
        	Error:      	Received unexpected error:
        	            	Error response from daemon: could not find plugin bridge in v1 plugin registry: plugin not found: failed to create container
        	Test:       	TestRabbitmqIntegration/Running_rabbitmq_3.9
--- FAIL: TestRabbitmqIntegration (0.00s)
    --- FAIL: TestRabbitmqIntegration/Running_rabbitmq_3.9 (15.28s)
```

**Comment by Frapschen** [comment]
probably miss the two line? 
```
//go:build integration
// +build integration
```

**Comment by Frapschen** [comment]
the only two pipline errors is not related

**Comment by Frapschen** [comment]
@djaglowski need review pls :)

**Review by djaglowski — APPROVED** [review]
Thanks @Frapschen!

**Comment by djaglowski** [comment]
```
Running target 'do-integration-tests-with-cover' in module 'receiver/rabbitmqreceiver' as part of group 'all'
make -C receiver/rabbitmqreceiver "do-integration-tests-with-cover"
make[1]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/rabbitmqreceiver'
running go integration test ./... + coverage in /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/rabbitmqreceiver
=== RUN   TestRabbitmqIntegration
=== RUN   TestRabbitmqIntegration/Running_rabbitmq_3.9
=== PAUSE TestRabbitmqIntegration/Running_rabbitmq_3.9
=== CONT  TestRabbitmqIntegration/Running_rabbitmq_3.9
2022/06/20 03:06:59 Starting container id: d4df58b33a08 image: docker.io/testcontainers/ryuk:0.3.3
2022/06/20 03:07:00 Waiting for container id d4df58b33a08 image: docker.io/testcontainers/ryuk:0.3.3
2022/06/20 03:07:00 Container is ready id: d4df58b33a08 image: docker.io/testcontainers/ryuk:0.3.3
2022/06/20 03:07:07 Starting container id: 58b571e5967d image: 4abcccfc-84aa-42c6-91eb-34d5643b1737:0d6407ff-3b19-4598-817c-64cacb4a27d5
2022/06/20 03:07:07 Waiting for container id 58b571e5967d image: 4abcccfc-84aa-42c6-91eb-34d5643b1737:0d6407ff-3b19-4598-817c-64cacb4a27d5
2022/06/20 03:07:13 Container is ready id: 58b571e5967d image: 4abcccfc-84aa-42c6-91eb-34d5643b1737:0d6407ff-3b19-4598-817c-64cacb4a27d5
2022/06/20 03:07:17 Starting container id: 58b571e5967d image: 4abcccfc-84aa-42c6-91eb-34d5643b1737:0d6407ff-3b19-4598-817c-64cacb4a27d5
2022/06/20 03:07:17 Waiting for container id 58b571e5967d image: 4abcccfc-84aa-42c6-91eb-34d5643b1737:0d6407ff-3b19-4598-817c-64cacb4a27d5
2022/06/20 03:07:17 Container is ready id: 58b571e5967d image: 4abcccfc-84aa-42c6-91eb-34d5643b1737:0d6407ff-3b19-4598-817c-64cacb4a27d5
--- PASS: TestRabbitmqIntegration (0.00s)
    --- PASS: TestRabbitmqIntegration/Running_rabbitmq_3.9 (28.71s)
PASS
coverage: 63.1% of statements
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/rabbitmqreceiver	28.[772](https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/6959785527?check_suite_focus=true#step:5:773)s	coverage: 63.1% of statements
```

**Pull request merged**